lib/Locale/Maketext/Test.pm

```diff
@@ -0,0 +1,565 @@
+Otherwise, if at least one of the parameters passed to %plural contains a %d,
+all of the parameters must contain the %d as well.
+
+That means the following 2 %plural calls are allowed in Russian:
+
+    %plural(%3,%d книга,%d книги,%d книг)
+    %3 %plural(%3,книга,книги,книг)
+
+while this is forbidden:
+
+    %plural(%3,одна книга,%d книги,%d книг)
+
+=head1 SUBROUTINES/METHODS
+
+=head2 debug
+
+set this if you need to check warnings along with errors
+
+=cut
+
+use Moose;
+use namespace::autoclean;
+
+has debug => (
+    is      => 'ro',
+    isa     => 'Bool',
+    default => 0
+);
+
+=head2 directory
+
+directory where locales files are located
+
+=cut
+
+has directory => (
+    is       => 'ro',
+    isa      => 'Str',
+    required => 1
+);
+
+=head2 languages
+
+language array, set this if you want to test specific language only in specified directory
+
+=cut
+
+has languages => (
+    is      => 'rw',
+    isa     => 'ArrayRef[Str]',
+    default => sub { [] });
+
+=head2 auto
+
+flag to fallback when a key is missing from lexicons
+
+    # if this is not set then maketext will output errors if
+    # translations is marked as fuzzy or is missing
+    # read more about it here https://metacpan.org/pod/Locale::Maketext::Lexicon
+    Locale::Maketext::Lexicon->import({ _auto => 1 })
+
+=cut
+
+has auto => (
+    is      => 'ro',
+    isa     => 'Bool',
+    default => 0
+);
+
+has _status => (
+    is       => 'rw',
+    isa      => 'HashRef',
+    lazy     => 1,
+    init_arg => undef,
+    default  => sub {
+        {
+            status   => 1,
+            errors   => {},
+            warnings => {}};
+    });
+
+=head2 BUILD
+
+=cut
+
+sub BUILD {
+    my $self = shift;
+
+    unless (scalar @{$self->languages}) {
+        my @lang = ();
+        if (opendir my $dh, $self->directory) {
+            while (readdir $dh) {
+                if (my ($x) = grep { $_ =~ /^(\w+)\.po$/ } $_) {
+                    push @lang, ($x) = split(/\./, $x);
+                }
+            }
+            @lang = sort @lang;
+        }
+        $self->languages(\@lang);
+    }
+
+    Locale::Maketext::ManyPluralForms->import({
+            '_decode' => 1,
+            '_auto'   => $self->auto,
+            '*'       => ['Gettext' => File::Spec->rel2abs($self->directory) . '/*.po']});
+}
+
+=head2 testlocales
+
+test po files in directory specified
+
+=cut
+
+sub testlocales {
+    my $self = shift;
+
+    foreach my $lang (@{$self->languages}) {
+        my $po  = $self->_get_po($lang);
+        my $lg  = $po->{header}->{language};
+        my $hnd = Locale::Maketext::ManyPluralForms->get_handle($lg);
+        $hnd->plural(1, 'test %d');
+
+        my $plural_sub = $hnd->{_plural};
+
+        my $nplurals = 2;    # default
+        $nplurals = $1 if $po->{header}->{'plural-forms'} =~ /\bnplurals=(\d+);/;
+        my @plural;
+
+        for (my ($i, $j) = (0, $nplurals); $i < 10000 && $j > 0; $i++) {
```

Comment:
@basiliscos its just a random number to capture all plurals, we can any lower number as well